frontend/apps/marketing/src/components/afeEligibility/AFEForm.tsx

```diff
@@ -0,0 +1,464 @@
+  };
+
+  const handleCheckboxChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const {name, checked} = e.target;
+    updateFormData({[name]: checked});
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    // Let the browser show validation messages
+    if (!e.currentTarget.checkValidity()) return;
+
+    setFormSubmitting(true);
+
+    const formJson = JSON.stringify(formData);
+    sessionStorage?.setItem(SESSION_STORAGE_KEY, formJson);
+
+    if (!isSignedIn) {
+      logEvent('AFE Continue', undefined, {
+        submitData: formJson,
+        isSignedIn: String(isSignedIn),
+      });
+
+      setFormSubmitting(false);
+      setShowSubmissionNotice(true);
+
+      return;
+    }
+
+    logEvent('AFE Submit', undefined, {
+      formEmail: formData.email,
+      formSchoolId: formData.schoolId,
+      formData: formJson,
+    });
+
+    fetch(getStudioUrl('/dashboardapi/v1/amazon_future_engineer_submit'), {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      credentials: 'include',
+      body: new URLSearchParams({
+        'amazon_future_engineer[email]': formData.email,
+        'amazon_future_engineer[firstName]': formData.firstName,
+        'amazon_future_engineer[lastName]': formData.lastName,
+        'amazon_future_engineer[schoolId]': formData.schoolId,
+        'amazon_future_engineer[primaryProfessionalRole]':
+          formData.professionalRole,
+        'amazon_future_engineer[gradesTeaching]':
+          formData.gradeBands.join(', '),
+        'amazon_future_engineer[inspirationKit]': String(
+          formData.inspirationKit,
+        ),
+        'amazon_future_engineer[csta]': String(formData.consentCSTA),
+        'amazon_future_engineer[consentCSTA]': String(formData.consentCSTA),
+        'amazon_future_engineer[consentAFE]': String(formData.consentAFE),
+      }),
+    })
+      .then(async response => {
+        if (response.ok) {
+          sessionStorage?.removeItem(SESSION_STORAGE_KEY);
+          setShowSubmissionNotice(true);
+          return true;
+        } else {
+          const errorText = await response.text();
+          throw new Error(
+            `Form submission failed with HTTP ${response.status} ${response.statusText}: ${errorText}`,
+          );
+        }
+      })
+      .catch(error => {
+        setFormError(
+          typeof error === 'string'
+            ? error
+            : 'Something went wrong. Please try again later',
+        );
+      })
+      .finally(() => {
+        setFormSubmitting(false);
+      });
+  };
+
+  useEffect(() => {
+    if (isSignedIn && formRef.current?.checkValidity()) {
+      formRef.current.requestSubmit();
+    }
+  }, [isSignedIn]);
+
+  useEffect(() => {
+    updateFormData({email: email});
+  }, [email]);
+
+  useEffect(() => {
+    updateFormData({schoolId, schoolName});
+  }, [schoolId]);
```

Comment:
nit: This will not update if schoolName changes but schoolId stays the same.  Can we also add schoolName to dependencies please?